oss-licenses-plugin/src/main/groovy/com/google/android/gms/oss/licenses/plugin/OssLicensesPlugin.groovy

```diff
@@ -50,7 +50,7 @@ class OssLicensesPlugin implements Plugin<Project> {
         licenseTask.dependsOn(getDependencies)
 
         project.android.applicationVariants.all { BaseVariant variant ->
-            variant.preBuild.dependsOn(licenseTask)
+            variant.preBuildProvider.get().dependsOn(licenseTask)
```

Comment:
1. `variant.preBuildProvider` requires AGP 3.3+, you should provide a fallback for older versions.
2. Calling `Provider.get()` will cause the task to be prematurely configured.
I think the correct way would be:
```groovy
variant.preBuildProvider.configure { dependsOn(licenseTask) }
```

Reply:
Thanks for the review! Do you know where I can see `variant.preBuildProvider` requires AGP 3.3? I was looking for documentation on this, but couldn't find any.

Reply:
The release notes for 3.3 mention that the plugin started using new lazy task configuration APIs
https://developer.android.com/studio/releases/gradle-plugin#behavior_changes